Stack/Opc.Ua.Core/Types/Utils/HiResClock.cs

```diff
@@ -20,47 +20,40 @@ namespace Opc.Ua
     /// <summary>
     /// Produces high resolution timestamps.
     /// </summary>
-    public class HiResClock
+    public static class HiResClock
     {
-        /// <summary>
-        /// Returns the current UTC time (bugs in HALs on some computers can result in time jumping backwards).
-        /// </summary>
-        public static DateTime UtcNow
+        // In Net Core 3.0+ has monotonic timer from system start.
+        // https://docs.microsoft.com/ru-ru/dotnet/api/system.environment.tickcount64?view=netcore-3.1
+#if NETCORE3_0 || NETCORE3_1
+        public static long TickCount64 { get { return Environment.TickCount64; } } 
+#else
+
+        /// <summary>Get monotonic timer from system starts in Windows in Msec. In other systems using DateTime. (posible lost monitored items when combuter time shift. see #1121)</summary>
+        public static long TickCount64
         {
             get
             {
-                return DateTime.UtcNow;                
+                if (Environment.OSVersion.Platform == PlatformID.Win32NT)
```

Comment:
Better hardcode #if NETFRAMEWORK? But we still support mono builds...this function is called quite function, maybe you can do something like for Utils.IsRunningOnMono, a lazy static variable.